src/codegen/insert/insert_scan_translator.cpp

```diff
@@ -0,0 +1,151 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// hash_group_by_translator.cpp
+//
+// Identification: src/codegen/insert/insert_scan_translator.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "codegen/pool/pool_runtime_proxy.h"
+#include "codegen/raw_tuple/raw_tuple_runtime_proxy.h"
+#include "codegen/insert/insert_helpers_proxy.h"
+#include "codegen/schema/schema_proxy.h"
+#include "codegen/catalog_proxy.h"
+#include "codegen/data_table_proxy.h"
+#include "planner/abstract_scan_plan.h"
+#include "storage/data_table.h"
+#include "codegen/insert/insert_scan_translator.h"
+#include "codegen/raw_tuple/raw_tuple_ref.h"
+
+namespace peloton {
+namespace codegen {
+
+InsertScanTranslator::InsertScanTranslator(const planner::InsertPlan &insert_plan,
```

Comment:
I understand this insert translator assumes it has a child of scan .... but the class name is a bit confusing at first glance